src/Features/LanguageServer/Protocol/Handler/CodeActions/CodeActionHelpers.cs

```diff
@@ -166,17 +164,15 @@ internal static class CodeActionHelpers
         {
             var actionSets = await GetActionSetsAsync(
                 document, codeFixService, codeRefactoringService, selection, cancellationToken).ConfigureAwait(false);
-            if (!actionSets.HasValue)
+            if (actionSets.IsDefaultOrEmpty)
```

Comment:
⚠️ This will prevent `GetActionSetsAsync` from using the cached `Task<TResult>` if it happens to complete synchronously.

Reply:
i'm missing something.  can you clarify what you mean by that?